pkg/strategy/grid2/strategy.go

```diff
@@ -875,48 +875,48 @@ func (s *Strategy) cancelAll(ctx context.Context) error {
 	if s.UseCancelAllOrdersApiWhenClose {
 		s.logger.Infof("useCancelAllOrdersApiWhenClose is set, using advanced order cancel api for canceling...")
 
-		if s.OrderGroupID > 0 {
-			s.logger.Infof("found OrderGroupID (%d), using group ID for canceling orders...", s.OrderGroupID)
+		for {
+			s.logger.Infof("checking %s open orders...", s.Symbol)
 
-			op := func() error {
-				_, cancelErr := service.CancelOrdersByGroupID(ctx, s.OrderGroupID)
-				return cancelErr
-			}
-			err := backoff.Retry(op, backoff.WithMaxRetries(backoff.NewExponentialBackOff(), 101))
+			openOrders, err := session.Exchange.QueryOpenOrders(ctx, s.Symbol)
 			if err != nil {
-				werr = multierr.Append(werr, err)
-			}
-		} else {
-			s.logger.Infof("canceling all orders...")
-			op := func() error {
-				_, cancelErr := service.CancelAllOrders(ctx)
-				return cancelErr
+				return err
```

Comment:
Could it fallback to cancel all orders anyway if having any error? Querying open orders is only for skipping when there is no open orders.

Reply:
it's exp backoff, so if we have error, it will always retry